### PR TITLE
Flake8  —select=E999 .

### PR DESCRIPTION
**Pull request by cclauss**
flake8 is a great way to find issues in Python code.  E999 issues are syntax errors which are often things that work in Python 2 but not in Python 3.  I made changes so that code will run in both Py2 and Py3.

### CODE REVIEW

File: core/actionProxy/actionproxy.py [hunk 1]
```diff
@@ -46,59 +47,50 @@ def __init__(self, source = None, binary = None):
     # @return True iff binary exists and is executable
     def init(self, message):
         def prep():
-            if 'code' in message:
+            if 'code' in message and message['code'] is not None:
                 binary = message['binary'] if 'binary' in message else False
                 if not binary:
-                    with codecs.open(self.source, 'w', 'utf-8') as fp:
```
Comment (rabbah) [inline core/actionProxy/actionproxy.py]: did you rebase to master?

File: core/javaAction/init.py [hunk 2]
```diff
@@ -18,12 +15,12 @@
     encoded = sys.argv[1]
 
 payload = {
```
Comment (rabbah) [inline core/javaAction/init.py]: we may delete this script now (with this change https://github.com/openwhisk/openwhisk/pull/1938/files#diff-80a0161d4fae50b554066617d1de0239R83, i think the `invoke.py` script in `actionproxy` will work for all the runtimes although i did not verify).

File: core/pythonAction/cli/wskactivation.py [hunk 3]
```diff
@@ -21,7 +21,13 @@
 from datetime import datetime, timedelta
 import copy
 from wskitem import Item
-from wskutil import addAuthenticatedCommand, apiBase, bold, parseQName, getQName, request, responseError, getPrettyJson
```
Comment (rabbah) [inline core/pythonAction/cli/wskactivation.py]: all this source in pythonAction/cli we need to delete (these changes can may be replayed against the cli repo).
what to just delete this source (@mrutkows  fyi)

File: core/javaAction/init.py (hunk 2 above)
Comment (rabbah) [inline core/javaAction/init.py]: btw, https://github.com/openwhisk/openwhisk/blob/master/core/actionProxy/invoke.py#L66 should be >= 1 (mind fixing at the same time?)

**Comment by rabbah** [comment]
@cclauss thanks for making these changes.
we should consider adding to our `scanCode.py` script to apply these checks automatically (or update travis).
what do you think?

File: core/actionProxy/actionproxy.py (hunk 1 above)
Comment (cclauss) [inline core/actionProxy/actionproxy.py]: I struggle with git...  Sorry.

I would have rewritten this as `if message.get('code'):` which fails if 1) 'code' not in message, 2) message['code'] in [None, '', False, 0, 0.0, [], {}, set()]

Comment (rabbah) [inline core/actionProxy/actionproxy.py]: No don't - I'll rebase when merging. Hopefully conflicts are easy to resolve.

Comment (rabbah) [inline core/actionProxy/actionproxy.py]: Rewrite as you see fit - code improvements and best practices are always appreciated.

File: core/javaAction/init.py (hunk 2 above)
Comment (cclauss) [inline core/javaAction/init.py]: When running the actionProxy on localhost, do we want to allow `./invoke.py run` with no dictionary parameter after `run`?

**Comment by cclauss** [comment]
I put in the change to core/actionProxy/invoke.py requested above.

The flake8 checking is in https://github.com/openwhisk/openwhisk/pull/1954  Flake8 will do all the checks that `scanCode.py` does but flake8 will _only_ look at .py files.

**Comment by markusthoemmes** [comment]
@cclauss Can you collapse the 2 pull-requests? It feels a bit odd to first do changes to the files and later on add the tool to actually do the checks.

Otherwise thanks for doing this! This is goodness for our code-hygiene! :+1:

**Comment by cclauss** [comment]
@markusthoemmes The first PR was done by running flake8 locally on my Mac.  It is a self-contained bit of work.  The second was done to add flake8 testing to the repo so that it will be automatically  invoked on all future pull requests.  I am not a git master and most of the docs on merging (ex. https://help.github.com/enterprise/2.9/user/articles/about-pull-request-merges/) assume write access to the repo.

**Comment by rabbah** [comment]
@cclauss can you rebase this pr? You can do this by:

```
git fetch --all [--prune]
git rebase upstream/master
```

**Comment by cclauss** [comment]
Rebased into https://github.com/openwhisk/openwhisk/pull/1973

**Pull request closed without merging**